ocaml/libs/xapi-inventory/lib/inventory.ml

```diff
@@ -52,7 +52,7 @@ let inventory = Hashtbl.create 10
 
 let inventory_m = Mutex.create ()
 
-let new_uuid = Uuidm.v4_gen (Random.get_state ())
```

Comment:
This is not safe for OCaml 5 (in fact I think Uuidm's own documentation is wrong here, and hasn't been updated for OCaml 5). Sharing a State.t between domains is a bad idea.

Although this change doesn't introduce any new problems, because we've already had this problem before, we just didn't see it because it was hidden inside Uuidm.
(in fact strictly speaking Uuidm was only thread-safe previously if you ever called Uuidm on the main thread once to force the lazy value, because we know that on OCaml 4 forcing lazy values from multiple threads can lead to a segfault).